pkg/archive/archive_unix.go

```diff
@@ -88,8 +89,9 @@ func minor(device uint64) uint64 {
 // handleTarTypeBlockCharFifo is an OS-specific helper function used by
 // createTarFile to handle the following types of header: Block; Char; Fifo
 func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
-	if userns.RunningInUserNS() {
-		// cannot create a device if running in user namespace
+
+	if userns.RunningInUserNS() && hdr.Typeflag != tar.TypeFifo {
+		logrus.Warnf("Tar: Can't create device %v while running in user namespace", path)
```

Comment:
I'm not certain that we want to start warning about things we don't do when we're in a user namespace.
Looking for other instances where we did or didn't, it looks like we'll also need to tweak line 649 in archive.go, and maybe we end up dropping this block.

Reply:
Ok I will drop this to debug.

Reply:
Probably going to need to touch https://github.com/containers/storage/blob/main/pkg/archive/archive.go#L649 also.

Reply:
+1 for the warning. This issue hit me because I was running my app in user namespace and I couldn't figure out the problem why my fifos were'nt there.